framework/PageWizard/PageWizardBody.tsx

```diff
@@ -40,54 +38,55 @@ export function PageWizardBody<T>({
 
   const onNext = useCallback(
     async (formData: object) => {
-      const filteredSteps = allSteps.filter((step) =>
-        isStepVisible(step, { ...wizardData, ...formData })
-      );
-
       if (activeStep === null) {
         return Promise.resolve();
       }
 
-      if (activeStep.validate) {
+      if (!isPageWizardParentStep(activeStep) && activeStep.validate) {
         await activeStep.validate(formData, wizardData);
       }
 
-      const isLastStep = activeStep?.id === filteredSteps[filteredSteps.length - 1]?.id;
+      const isLastStep =
+        activeStep?.id === visibleStepsFlattened[visibleStepsFlattened.length - 1]?.id;
       if (isLastStep) {
         return onSubmit(wizardData as T);
       }
 
-      const activeStepIndex = filteredSteps.findIndex((step) => step.id === activeStep?.id);
-      const nextStep = filteredSteps[activeStepIndex + 1];
+      const activeStepIndex = visibleStepsFlattened.findIndex((step) => step.id === activeStep?.id);
+      // If the next step is a parent step, mark its first substep as the next active step
+      const nextStep = isPageWizardParentStep(visibleStepsFlattened[activeStepIndex + 1])
+        ? visibleStepsFlattened[activeStepIndex + 2]
```

Comment:
I think this may throw an error if substeps is an empty array. (Is there a way to ensure the substeps array is at least length one in the type definition?)

Reply:
Great catch! I updated the type for `substeps` to require at least one item and verified that TS throws an error on an empty array for `substeps`.